modules/account-lib/src/coin/sol/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BuildTransactionError } from '../baseCoin/errors';
+import { Transaction } from './transaction';
+import { isValidAmount, validateAddress } from './utils';
+import { TransactionType } from '../baseCoin';
+import { InstructionBuilderTypes } from './constants';
+import { TokenTransfer } from './iface';
+
+import assert from 'assert';
+import { TransactionBuilder } from './transactionBuilder';
+
+export interface SendParams {
+  address: string;
+  amount: string;
+  mintAddress: string;
+  ownerAddress: string;
+}
+
+export class TokenTransferBuilder extends TransactionBuilder {
+  private _sendParams: SendParams[] = [];
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.Send;
+  }
+
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+
+    for (const instruction of this._instructionsData) {
+      if (instruction.type === InstructionBuilderTypes.TokenTransfer) {
+        const transferInstruction: TokenTransfer = instruction;
+        this.sender(transferInstruction.params.fromAddress);
+        this.send({
+          address: transferInstruction.params.toAddress,
+          amount: transferInstruction.params.amount,
+          mintAddress: transferInstruction.params.mintAddress,
+          ownerAddress: transferInstruction.params.ownerAddress,
+        });
+      }
+    }
+  }
+
+  /**
+   *  Set a transfer
+   *
+   * @param {string} fromAddress - the sender address
+   * @param {string} toAddress - the receiver address
+   * @param {string} amount - the amount sent
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  send({ address, amount, mintAddress, ownerAddress }: SendParams): this {
+    validateAddress(address, 'address');
+    if (!amount || !isValidAmount(amount)) {
+      throw new BuildTransactionError('Invalid or missing amount, got: ' + amount);
+    }
+
+    this._sendParams.push({ address, amount, mintAddress: mintAddress, ownerAddress: ownerAddress });
+
+    return this;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    assert(this._sender, 'Sender must be set before building the transaction');
+
+    const transferData = this._sendParams.map((sendParams: SendParams): TokenTransfer => {
+      return {
+        type: InstructionBuilderTypes.TokenTransfer,
+        params: {
+          fromAddress: this._sender,
+          toAddress: sendParams.address,
+          amount: sendParams.amount,
+          mintAddress: sendParams.mintAddress,
+          ownerAddress: sendParams.ownerAddress,
```

Comment:
We shouldn't need to maintain a separate `ownerAddress` property. this should be the same as `fromAddress`.

Also, let me know how you feel about this approach to support multi token transfers:
1. `this._sender` should always be the wallet's root address (also `ownerAddress`).
2. An ATA's address is deterministically derived from `(ownerAddress, mintAddress)` - this should give us the actual `fromAddress`.

Reply:
Also double check this, but make sure the `feeAddress` is the native sol address (which i believe is taken care of automatically if you set it as `sender`)

Reply:
yep, this is inside buildSolTransaction: `tx.feePayer = new PublicKey(this._sender);`!